Code/GraphMol/RGroupDecomposition/Wrap/rdRGroupComposition.cpp

```diff
@@ -126,18 +124,19 @@ class RGroupDecompositionHelper {
     for (const auto &side_chains : groups) {
       python::dict dict;
       for (const auto &side_chain : side_chains) {
+        const auto &mol = side_chain.second;
         if (asSmiles) {
-          dict[side_chain.first] = MolToSmiles(*side_chain.second, true);
+          dict[side_chain.first] = MolToSmiles(*mol, true);
         } else {
-          dict[side_chain.first] = side_chain.second;
+          dict[side_chain.first] = mol;
         }
       }
```

Comment:
You could make this even cleaner as:
```
      for (const auto &[lbl, mol] : side_chains) {
        if (asSmiles) {
          dict[lbl] = MolToSmiles(*mol, true);
        } else {
          dict[lbl] = mol;
        }
      }
```

Reply:
Ah, right, destructuring to the rescue! That's much cleaner, I will update my PR with your suggestion.